solutions/숨바꼭질.js

```diff
@@ -0,0 +1,35 @@
+const INPUT_PATH = "./inputs/숨바꼭질.txt";
+
+const fs = require("fs");
+const filePath = process.platform === "linux" ? "/dev/stdin" : INPUT_PATH;
+const input = fs.readFileSync(filePath).toString().trim().split("\n");
+
+function solution(inputArguments) {
+  const [N, K] = inputArguments[0].split(" ").map(Number);
+
+  let time = 0;
+  const searchList =[[N, time]]
+
+  while (searchList.length > 0) {
+    const [current, accTime] = searchList.pop();
+
+    const leftMove = current - 1;
+    const rightMove = current + 1;
+    const doubleMove = current * 2;
+
+    if (current === K) {
+      return accTime;
+    }
+
+    if (current < K) {
+      searchList.push([rightMove, accTime + 1]);
+      searchList.push([doubleMove, accTime + 1]);
+    } else {
+      searchList.push([leftMove, accTime + 1]);
+    }
```

Comment:
```suggestion
    if (current < K) {
      searchList.push([rightMove, accTime + 1]);
      searchList.push([doubleMove, accTime + 1]);
    } else {
      searchList.push([leftMove, accTime + 1]);
    }
```

저도 다른분들 풀이 보면서 깨달았는데, `current`가 `K`보다 작아도 `left`로 움직여야 하는 경우가 있습니다!
따라서 `if`로 분기처리 해두신걸 빼면 풀릴 것 같아요!